lib/async_hooks.js

```diff
@@ -223,15 +224,20 @@ class AsyncResource {
     return this[trigger_async_id_symbol];
   }
 
-  bind(fn, thisArg = this) {
+  bind(fn, thisArg) {
     validateFunction(fn, 'fn');
-    const ret =
-      FunctionPrototypeBind(
-        this.runInAsyncScope,
-        this,
-        fn,
-        thisArg);
-    ObjectDefineProperties(ret, {
+    const runInAsyncScope = this.runInAsyncScope;
+    const resource = this;
+    let bound;
+    if (thisArg === undefined) {
```

Comment:
What if the user would like `thisArg` to be undefined? I think using `arguments.length` makes more sense here, I don't see a reason to not let `undefined` be a valid value for `thisArg`.

Reply:
This would prevent ever adding any new arguments without a breaking change to allow `undefined` as I'm doing right now. I think the tradeoff of `undefined` being a special value makes more sense.

For example:

```js
asyncResource.bind(fn, undefined, 'foo') // potential future API
```

Reply:
We can use `if (arguments.length > 1)` instead.

Reply:
the workaround is to wrap the function to give a special value to `this`:

```mjs
ar.bind(function (...args) {
  return fn.call(undefined, ...args);
}, undefined, 'foo');
```

Which seems ok, and is kind of what has to be done in the cases of default arguments anyway. Default arguments in JS are only on `undefined` so this seems fine.